### PR TITLE
Automatically update vcpkg on non-mac systems

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/pipelines/templates/steps/vcpkg.yml [hunk 1]
```diff
@@ -39,7 +39,27 @@ steps:
...
+  # image at a given time stamp and not moved forward until some update
+  # procedure is run.
+  - task: Bash@3
+    inputs:
+      targetType: inline
+      script: |
+        cd $VCPKG_INSTALLATION_ROOT
+        git pull
+
+        ./bootstrap-vcpkg.sh
```
Comment (vhvb1989) [inline eng/pipelines/templates/steps/vcpkg.yml:54]: for windows it is: `\bootstrap-vcpkg.bat`.   not sure if sh will be always compatible

File: eng/pipelines/templates/steps/vcpkg.yml [hunk 2]
```diff
@@ -39,7 +39,27 @@ steps:
...
+
+  # Update vcpkg and re-bootstrap on non-Mac OS'. Vcpkg is installed on the
+  # image at a given time stamp and not moved forward until some update
+  # procedure is run.
+  - task: Bash@3
+    inputs:
+      targetType: inline
+      script: |
+        cd $VCPKG_INSTALLATION_ROOT
+        git pull
```
Comment (vhvb1989) [inline eng/pipelines/templates/steps/vcpkg.yml]: git pull origin master
may be?

File: eng/pipelines/templates/steps/vcpkg.yml (hunk 1 above)
Comment (danieljurek) [reply eng/pipelines/templates/steps/vcpkg.yml:54]: We're running the command in `bash` for both platforms so the command will succeed in both cases.

File: eng/pipelines/templates/steps/vcpkg.yml (hunk 2 above)
Comment (ahsonkhan) [reply eng/pipelines/templates/steps/vcpkg.yml]: Are we sure the remote will always be `origin`?

File: eng/pipelines/templates/steps/vcpkg.yml [hunk 3]
```diff
@@ -39,7 +39,28 @@ steps:
...
+        cd $VCPKG_INSTALLATION_ROOT
+
+        # Sometimes the image has changes in the git tree. Reset to HEAD and
+        # then pull
+        git reset --hard HEAD
+        git pull origin master
+
+        ./bootstrap-vcpkg.sh
+    condition: and(succeeded(), not(contains(variables['OSVmImage'], 'macOS')), not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')))
+    displayName: Upgrade vcpkg
```
Comment (ahsonkhan) [inline eng/pipelines/templates/steps/vcpkg.yml:60]: Is this the right display name or should it be called `vcpkg bootstrap` like before?
I don't see how we are upgrading here.

**Pull request merged**